syntax/impls.rs

```diff
@@ -274,3 +279,84 @@ impl Borrow<Type> for &Impl {
         &self.ty
     }
 }
+
+impl Pair {
+    /// Use this constructor when the item can't have a different
+    /// name in Rust and C++. For cases where #[rust_name] and similar
+    /// attributes can be used, construct the object by hand.
```

Comment:
nit: This feels a bit weird to me interface-wise; I think it would be clearer to have one way to build these. WDYT of having new() take two separate Ident args, use that everywhere rather than directly building the Pair, and then any places where you use this currently can just clone the ident when calling this. That feels clearer than having some places use Pair::new and other places build the Pair directly.

Reply:
I feel the underlying problem here is that we should consistently allow `#[rust_name]` and/or `#[cxx_name]` everywhere we declare any kind of name at all, and therefore (as you say) `Pair::new` will always take (potentially) different `Ident`s for Rust and C++. I didn't want to go that far, primarily because I didn't fancy adding 100 extra test cases :) #379 may inch slightly further in this direction.

But you're right the interface is no good. I have gone for a compromise by adding an extra constructor for the other case.